### PR TITLE
colw/Basic validator filtering by name

### LINKED ISSUE #2932: Add search to validators list
The validators list is long. Often users just want to look for one special validator. Let's give them a searchbar to search for validators.

AC:
- searchbar on the top of the validators list
- instant search on the searchbar using graphql filters

Optional:
- filter out jailed validators for now

### PR DESCRIPTION
**Pull request by colw**
Closes #2932 

**Description:**

I have create a new `PageContainer` component. It's `TmPage` without a lot of cruft that is less necessary now we use GraphQL etc.

Desktop:
Screenshot 2019-09-06 at 15 01 49

Mobile:
Screenshot 2019-09-06 at 15 01 38

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by jbibla** [comment]
do we need to keep `TmPage`?

**Comment by colw** [comment]
> do we need to keep `TmPage`?

I plan (would like) to get rid of it. But it would fall under another PR.

File: src/components/staking/PageValidators.vue [hunk 1]
```diff
@@ -8,31 +8,40 @@
     hide-header
   >
     <template slot="managed-body">
+      <TmField v-model="searchTerm" placeholder="Validator name…" />
```
Comment (jbibla) [inline src/components/staking/PageValidators.vue]:
```suggestion
      <TmField v-model="searchTerm" placeholder="Search" />
```

**Comment by jbibla** [comment]
this is awesome. i'm sorry i didn't provide a design for this in advance. but here are a few style notes:

- the search bar should have `1rem` of margin around it on mobile and `2rem` on desktop.
- can we set a max-width of 300px on tablet and up?
- size 14 font
- can we include a message for the empty state?

Screen Shot 2019-09-05 at 10 05 45 AM

**Comment by colw** [comment]
Have a look now @jbibla . See what you think.

**Comment by colw** [comment]
It isn't obvious anything really happens when you set show active only. need to figure that one.

File: src/components/common/PageContainer.vue [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+</template>
+
+<script>
+import TmPageHeader from "./TmPageHeader.vue"
+import { mapState, mapGetters } from "vuex"
+import TmBalance from "common/TmBalance"
+import PageFooter from "common/TmPageFooter"
+
+export default {
+  name: `tm-page`,
```
Comment (faboweb) [inline src/components/common/PageContainer.vue]:
```suggestion
  name: `page`,
```

Comment (colw) [reply src/components/common/PageContainer.vue]: Might make it `page-container` for consistency

**Comment by colw** [comment]
I added some basic transitions. They look a bit off right now.

**Comment by colw** [comment]
Need feedback on the button and transitions

**Comment by faboweb** [comment]
Doesn't show?

image

**Comment by colw** [comment]
Hmm the Netlify deploy . has an old version present. Locally it's fine for me.

Want to check it out locally?

**Comment by colw** [comment]
Screenshot 2019-09-05 at 19 59 31

That's what I see when I click the Netlify build. Which looks like the first version. Is Netlify not rebuilding on each push?

**Comment by faboweb** [comment]
image

Transitions are nice. Searchbar and button look asymetrical and out of place to me. What does mr designer say? @jbibla

**Comment by faboweb** [comment]
It should rebuild every push. Weird.

**Comment by colw** [comment]
> image
> 
> Transitions are nice. Searchbar and button look asymetrical and out of place to me. What does mr designer say? @jbibla

Agree about the buttons. I left things standard.

We didn't have a toggle, so I just used a button. I put it underneath to avoid it looking like a "search" button. Would prefer @jbibla input before going down too far any paths.

**Comment by colw** [comment]
Updated. Still a bit dull, but consistent.

Screenshot 2019-09-06 at 12 12 32

**Comment by colw** [comment]
Updated again. See top @jbibla

**Pull request merged**